agent/config/builder.go

```diff
@@ -1318,6 +1318,7 @@ func (b *Builder) svcTaggedAddresses(v map[string]ServiceAddress) map[string]str
 }
 
 func (b *Builder) serviceVal(v *ServiceDefinition) *structs.ServiceDefinition {
+	var validDNSre = regexp.MustCompile(`^[a-zA-Z0-9]([a-zA-Z0-9\-]{0,62}[a-zA-Z0-9])?$`)
```

Comment:
There are some schools of thought that suggest putting all variable declarations at the start of a function, but personally I follow the school that suggests declaring them immediately before the first use. Looking at some of the other declarations in this function I think this code base may follow the "immediately before first use" philosophy.

My rational for keeping the variable declaration immediately before the first use is:
1. it reduces the scope of the variable, which makes it easier to reason about. Code before the variable is declared can't use or modify the value
2. it makes it easier to refactor the code later because you can pull out a single block of code, rather than pulling out lines from different places in the function.